.github/workflows/test.yml

```diff
@@ -1,14 +1,25 @@
 name: test
-on: [ push, pull_request ]
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 env:
   UNDERCOVER_CONFIG: "(\"elsa.el\" \"elsa-*.el\" (:report-format lcov) (:send-report nil))"
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
Maybe format this one per line so it is a bit visually clearer.

Reply:
Of course! Done!